updated doc to build arm/arm64 versions of psrp.windows library

SteveL-MSFT opened the pull request:
## PR Summary
The doc to build and publish psrp.windows library needed to be updated to support arm and arm64 builds.  It was also missing steps for signing and publishing.  Added details on individual features needed for VS2017 and location to publish on MyGet.

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [X] Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed - Issue link:
- [NA] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [ ] [Add `[feature]` if the change is significant or affectes feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

SteveL-MSFT committed:
updated doc to build arm/arm64 versions of psrp.windows library

daxian-dbw commented on docs/building/internals.md:
```diff
@@ -115,22 +115,39 @@ and the NuGet package for `libpsl-native` is `libpsl`.
+The `x64_arm` and `x64_arm64` architectures mean that the host system needs to be x64 to cross-compile to ARM.
+When building for multiple architectures, be sure to use the `-clean` switch as cmake will cache the previous run and the wrong compiler will be used to generate the subsequent architecutes.
+
+After that, the binary `pwrshplugin.dll` and its PDB file will be placed under 'src\powershell-native\bin\release\coreclr'.
```
Maybe we should use the path 'src\powershell-win-core' here. The dll and pdb files will be bin-placed to "src\powershell-win-core" (see the [script](https://github.com/PowerShell/PowerShell/blob/master/build.psm1#L277)), and it's the same to 'PowerShell.Core.Instrumentation.dll' which will also be produced when running `Start-BuildNativeWindowsBinaries`.

SteveL-MSFT replied:
Ok, will change it.

daxian-dbw commented on docs/building/internals.md:
````diff
@@ -142,13 +159,24 @@ The layout of files should look like this:
 ```
 
-Lastly, run `nuget pack .` from within the folder. Note that you may need the latest `nuget.exe`.
+Have the dlls signed and run `nuget pack` from the parent of the `runtimes` folder where `psrp.windows.nuspec` resides.
````
> Have the dlls signed

 Please add "with `authenticode dual` certificate"

SteveL-MSFT replied:
Will add.

daxian-dbw commented:
@SteveL-MSFT  I just released we don't have arm32/arm64 version of `PowerShell.Core.Instrumentation.dll` for the [`PowerShell.Core.Instrumentation`](https://powershell.myget.org/feed/powershell-core/package/nuget/PowerShell.Core.Instrumentation/6.0.0-beta.10) nuget package ...

Can we build `PowerShell.Core.Instrumentation.dll` with `x64_arm64` and `x64_arm`?

SteveL-MSFT commented:
@daxian-dbw I'll publish a new pkg for those dlls

SteveL-MSFT commented:
I'll merge the psrp.windows and powershell.core.instrumentation sections into a general Windows native section

SteveL-MSFT committed:
fix spelling
address PR feedback
merge PowerShell.Core.Instrumentation section into PSRP.Windows section

SteveL-MSFT commented:
Published https://powershell.myget.org/feed/powershell-core/package/nuget/PowerShell.Core.Instrumentation/6.0.0-RC2
